pdns/dnsdist.cc

```diff
@@ -1885,9 +1885,10 @@ static void setUpLocalBind(std::unique_ptr<ClientState>& cs)
 
   if(!cs->tcp && IsAnyAddress(cs->local)) {
     int one=1;
-    setsockopt(fd, IPPROTO_IP, GEN_IP_PKTINFO, &one, sizeof(one));     // linux supports this, so why not - might fail on other systems
+    (void)setsockopt(fd, IPPROTO_IP, GEN_IP_PKTINFO, &one, sizeof(one)); // linux supports this, so why not - might fail on other systems
 #ifdef IPV6_RECVPKTINFO
-    setsockopt(fd, IPPROTO_IPV6, IPV6_RECVPKTINFO, &one, sizeof(one));
+    if (setsockopt(fd, IPPROTO_IPV6, IPV6_RECVPKTINFO, &one, sizeof(one)) < 0)
+        warnlog("Warning: IPV6_RECVPKTINFO setsockopt failed: %s", stringerror());
```

Comment:
I don't really see the point of warning about this. There is nothing the user can do about it anyway.

Reply:
```
18:52 < cmouse> rgacogne: what happens if setting IPV6_RECVPKTINFO fails?
18:53 < cmouse> rgacogne: will it break anything? does something stop working?
19:00 < rgacogne> cmouse: we might not be able to properly detect the destination address for datagrams received over IPv6 for sockets bound to ::
19:01 < cmouse> rgacogne: point being; if setting that fails, will it cause something that might concern an admin?
19:01 < Habbie> yes
19:01 < Habbie> if they have :: in local-address
19:01 < cmouse> then I would suggest that warning would be logged
19:01 < Habbie> but
19:02 < Habbie> we really only want that warning once
19:02 < rgacogne> Habbie: it's for dnsdist in that case
19:02 < Habbie> oh
19:02 < rgacogne> anyway there is nothing an admin can do
19:03 < Habbie> except stop listening on ::
19:03 < Habbie> like the pdns faq used to warn
19:03 < rgacogne> and we would log that every time we start
19:03 < rgacogne> true
19:03 < Habbie> but, when would this happen? on some obsecure solaris fork?
19:03 < cmouse> if kernel does not support that option, most likely.
```

so what do we do?

Reply:
Warn once? (maybe have a (`static`?) `bool` to make sure it really is once.)

Reply:
Warning only once would be nice but please keep it simple, I don't care that much :-)

Reply:
Fixed.